server/hotrod/src/main/java/org/infinispan/server/hotrod/Decoder2x.java

```diff
@@ -411,6 +411,13 @@ public void customReadKey(HotRodHeader header, ByteBuf buffer, CacheDecodeContex
             }).isPresent()) {
                return;
             }
+            if (Constants.isVersionPost25(header.version)) {
+               int listenerInterests = ExtendedByteBufJava.readMaybeVInt(buffer);
+               if (listenerInterests == Integer.MIN_VALUE)
+                  return;
+
+               requestCtx.setListenerInterests(listenerInterests);
+            }
```

Comment:
We need to move `out.add(hrCtx);` from line 408 to after this if block so that if we don't have all bytes it works properly.  We also would need to mark the reader index before that.

I believe the tests should have caught this unless they aren't using the `SingleByteFrameDecoderChannelInitializer` correctly.